server/api/core/hasPermission.app-test.js

```diff
@@ -0,0 +1,68 @@
+import { Meteor } from "meteor/meteor";
+import { expect } from "meteor/practicalmeteor:chai";
+import { sinon } from "meteor/practicalmeteor:sinon";
+import { Factory } from "meteor/dburles:factory";
+import Core from "./core";
+
+describe("hasPermission", function () {
+  let sandbox;
+
+  beforeEach(function () {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  it("should return false when not passed any permissions", function () {
+    sandbox.stub(Meteor, "userId", () => "random-user-id");
+    const hasPermissions = Core.hasPermission();
+    expect(hasPermissions).to.be.false;
+  });
+
+  it("should return false when `permissions` is an empty string", function () {
+    sandbox.stub(Meteor, "userId", () => "random-user-id");
+    const hasPermissions = Core.hasPermission("");
+    expect(hasPermissions).to.be.false;
+  });
+
+  it("should return false when `permissions` is an empty array", function () {
+    sandbox.stub(Meteor, "userId", () => "random-user-id");
+    const hasPermissions = Core.hasPermission([]);
+    expect(hasPermissions).to.be.false;
+  });
+
+  it("should return false for users that don't have the `owner` permission", function () {
+    const registeredUser = Factory.create("registeredUser");
+    sandbox.stub(Meteor, "userId", () => registeredUser._id);
+    const hasPermissions = Core.hasPermission(["owner"]);
+    expect(hasPermissions).to.be.false;
+  });
+
+  it("should return true for anonymous users when `permissions === ['anonymous', 'guest']`", function () {
+    const user = Factory.create("anonymous");
+    sandbox.stub(Meteor, "userId", () => user._id);
+
+    // It is necessary to explicity supply shopId to the Core.hasPermission call
+    // below. If we don't, the said function defaults to using Core.getShopId
+    // which, when this test is run, will give a shop ID that will be different
+    // from the one used by Factory.create("anonymous").
+    const shopId = Object.keys(user.roles)[0];
```

Comment:
I'm not sure it's possible for that index to be 1. I based my decision to use index 0 off of the declaration of `Factory.define("anonymous")` in [this file](https://github.com/reactioncommerce/reaction/blob/b97b0757d7465e17166aaf7142fc0a2e829829e2/server/imports/fixtures/users.js). More specifically, if you check [this line](https://github.com/reactioncommerce/reaction/blob/b97b0757d7465e17166aaf7142fc0a2e829829e2/server/imports/fixtures/users.js#L102) in that file, you will see that `user.roles` can only have one key in it.

So, I do think this is the best approach. Or do you have another approach in mind?

Reply:
I meant that the fixture can change. A time will come when we make the fixtures multi-shop aware, then there will be more shopId keys (0,1 etc) there

Reply:
I think it will be better to wait till that change is made then. Coz until then, we can't know the structure of the new fixtures and the index of the right shopId to use above.